lib/RequestHelper.php

```diff
@@ -13,52 +13,106 @@
 
 namespace OCA\Files_Antivirus;
 
+use OC\Cache\CappedMemoryCache;
+use OC\Files\Storage\Storage;
 use \OCP\IRequest;
 
+/**
+ * Used to detect the size of the uploaded file
+ *
+ * @package OCA\Files_Antivirus
+ */
 class RequestHelper {
-
 	/**
 	 * @var  IRequest
 	 */
 	private $request;
 
+	/**
+	 * @var CappedMemoryCache
+	 */
+	private static $fileSizeCache;
+
+	/**
+	 * RequestHelper constructor.
+	 *
+	 * @param IRequest $request
+	 */
 	public function __construct(IRequest $request) {
 		$this->request = $request;
 	}
 
+	/**
+	 * @return CappedMemoryCache
+	 */
+	public function getCache() {
+		if (self::$fileSizeCache === null) {
+			self::$fileSizeCache = new CappedMemoryCache();
+		}
+		return self::$fileSizeCache;
+	}
+
+	/**
+	 * @param string $path
+	 * @param string $size
+	 *
+	 * @return void
+	 */
+	public function setSizeForPath($path, $size) {
+		$this->getCache()->set($path, $size);
+	}
+
 	/**
 	 * Get current upload size
 	 * returns null for chunks and when there is no upload
 	 *
+	 * @param Storage $storage
 	 * @param string $path
 	 *
 	 * @return int|null
 	 */
-	public function getUploadSize($path) {
-		$uploadSize = null;
+	public function getUploadSize(Storage $storage, $path) {
+		// remove .ocTransferId444531916.part from part files
+		$cleanedPath = preg_replace(
+			'|\.ocTransferId\d+\.part$|',
+			'',
+			$path
+		);
+		// cache uses dav path in /files/$user/path format
+		$translatedPath = preg_replace(
+			'|^files/|',
+			'files/' . $storage->getOwner('/') . '/',
+			$cleanedPath
+		);
+		$cachedSize = $this->getCache()->get($translatedPath);
+		if ($cachedSize > 0) {
+			return $cachedSize;
+		}
 
 		$requestMethod = $this->request->getMethod();
+		// Are we uploading anything?
+		if ($requestMethod !== 'PUT') {
+			return null;
+		}
 		$isRemoteScript = $this->isScriptName('remote.php');
 		$isPublicScript = $this->isScriptName('public.php');
-		// Are we uploading anything?
-		if (\in_array($requestMethod, ['MOVE', 'PUT']) && $isRemoteScript) {
+		if (!$isRemoteScript && !$isPublicScript) {
+			return null;
+		}
+
+		if ($isRemoteScript) {
 			// v1 && v2 Chunks are not scanned
-			if ($requestMethod === 'PUT' &&  \strpos($path, 'uploads/') === 0) {
+			if (\strpos($path, 'uploads/') === 0) {
 				return null;
 			}
 
-			if (\OC_FileChunking::isWebdavChunk() && \strpos($path, 'cache/') === 0) {
+			if (\OC_FileChunking::isWebdavChunk()
+				&& \strpos($path, 'cache/') === 0
+			) {
 				return null;
 			}
-
-			if ($requestMethod === 'PUT') {
-				$uploadSize = (int)$this->request->getHeader('CONTENT_LENGTH');
-			} else {
-				$uploadSize = (int)$this->request->getHeader('OC_TOTAL_LENGTH');
-			}
-		} else if ($requestMethod === 'PUT' && $isPublicScript) {
-			$uploadSize = (int)$this->request->getHeader('CONTENT_LENGTH');
 		}
+		$uploadSize = (int)$this->request->getHeader('CONTENT_LENGTH');
```

Comment:
are we not using OC_TOTAL_LENGTH for the final MOVE in new chunking ?

Reply:
@PVince81 it's used but could be omited https://github.com/owncloud/core/blob/master/apps/dav/lib/Upload/ChunkingPlugin.php#L97

Reply:
Well, I answered your question correctly, but neither the question not the answer is applicable here. This part is for non-chunked upload as the new chunking is handled with a Dav plugin above.